enhancements/20250820-support-multipath.md

```diff
@@ -0,0 +1,290 @@
+# Support Multipath Disk
+
+Support for multipath disk devices in Harvester to enable reliable storage access through multiple paths for improved fault tolerance and performance.
+
+## Summary
+
+This enhancement introduces multipath device recognition and management in Harvester's node-disk-manager component, allowing users to configure and utilize multipath storage devices safely while ensuring proper identification and avoiding conflicts with existing Longhorn v2 and LVM DM-devices.
+
+### Related Issues
+
+- Primary Issue: https://github.com/harvester/harvester/issues/6975
+- Non-WWN Device Handling Issue: https://github.com/harvester/harvester/issues/6261 (out of scope for this enhancement)
+
+## Motivation
+
+### Goals
+
+- Enable Harvester to recognize and manage multipath disk devices properly
+- Support multipath device configuration in Harvester installer with extended configuration options
+- Distinguish between multipath devices and Longhorn v2 DM-devices to avoid conflicts
+- Ensure multipath devices are back after starting multipathd service when multipathd service is disabled.
+
+### Non-goals
+
+- Implementing multipath configuration without DM_WWN support (this will be addressed in a separate issue #6261)
+
+## Proposal
+
+This enhancement adds multipath device support to Harvester by implementing device identification capabilities in the node-disk-manager component and extending the installer configuration options.
+
+### User Stories
+
+#### Story 1: Be able to use external storage like SAN, iSCSI, or Fibre Channel in NDM
+
+Those disks expose different accessible paths to the node. However, even we aggregate those paths in to a single device with multipath service, NDM doesn't still support recognizing them. Right now, it's only able to be configured in Longhorn. That's why we need to support configuaring in Harvester.
+
+After NDM is able to identify those aggregated disks, users can directly select those disk in Harvester instead of crating the settings in Longhorn.
+
+#### Story 2: High availability and fault tolerance
+
+Without multipath support, users cannot configure Harvester with multipath storage devices. This limitation results in single points of failure in storage connectivity and prevents effective use of enterprise SAN storage.
+
+With multipath, storage paths can be aggregated into a single logical device, enabling high availability and fault tolerance. In this setup, users can configure multipath devices during Harvester installation, and the system will automatically fail over to alternative paths if one becomes unavailable.
```

Comment:
Besides day 1 (installation time), how about day 2 at runtime?

Reply:
I meant below. In fact, users should be able to configure devices at runtime after installation?

>  In this setup, users can configure multipath devices during Harvester installation,